File: src/ProtectManager/component/PM_GUI/src/service/console/src/app/shared/httpInterceptors/http.params.ts
/*
* This file is a part of the open-eBackup project.
* This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
* If a copy of the MPL was not distributed with this file, You can obtain one at
* http://mozilla.org/MPL/2.0/.
*
* Copyright (c) [2024] Huawei Technologies Co.,Ltd.
*
* THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
* EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
* MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
*/
export class HttpExtParams {
  // 统一URL前缀
  akPrefix: string | 'none';

  // 是否显示loading效果
  akLoading = true;

  // 超时时间，默认30分钟
  akTimeout: number = 30 * 60 * 1e3;

  // 是否使用公共异常
  akDoException = true;

  // 是否逃逸注销
  akEscapeSession = false;
}

File: src/Infrastructure_OM/cbb_python/public_cbb/device_manager/http_params.py
# This file is a part of the open-eBackup project.
# This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
# If a copy of the MPL was not distributed with this file, You can obtain one at
# http://mozilla.org/MPL/2.0/.
#
# Copyright (c) [2024] Huawei Technologies Co.,Ltd.
#
# THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
# EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
# MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
#!/usr/bin/env python
# _*_ coding:utf-8 _*_

class SessionInfo:
    def __init__(
        self,
        base_url: str = '',
        device_id: str = '',
        token: str = '',
        cookie: str = '',
    ):
        self.base_url = base_url
        self.device_id = device_id
        self.token = token
        self.cookie = cookie
